### DISCUSSION TITLE
service user / fine grained user api access

### DISCUSSION THREAD

**Discussion by aheissenberger** [zitadel/zitadel#7805] [Q&A] [answered]
How can I allow a service user to only modify specific meta keys and roles for a user?

**UseCase:**
* one app in a project in an organization
* users from multiple organization can login
* the app needs  to change a few meta values and add/remove roles on a user based on his status
* the app should only be able to change the meta values and roles for a user with verified token

The only solution I found was to create a service user under the organization which holds the project. Then grant this user in all other organizations the role "ORG_USER_MANAGER". 
This gives the app more than it needs and opens up access to all users in the instance :-(

**Comment by hifabienne** [comment]
hei @aheissenberger 
We do have different manager roles which do allow different things.
I am not completely sure what exactly you need, but you can find the roles here: https://zitadel.com/docs/guides/manage/console/managers#roles

behind each role we have permissons such as user.read, project.read, project.write, etc
this defines which endpoints the user is allowed to perform a request or not.

if you are running on a selfhosted zitadel, you can add other roles with permissions to the defaults.yaml
In there all the roles with the permissions are defined

**Reply by aheissenberger** [reply]
1. this use case is not possible with the Zitadel cloud - there are no other options?
2. Is there any documentation of the permissions? I only found [this list ](https://zitadel.com/docs/self-hosting/manage/configure)as part of default config for self hosting. But no explanation what permission is needed to allow changing meta values.

**Reply by hifabienne** [reply, marked as answer]
At the moment it is not possible to define your own roles with permissions in the cloud. this is something we will probably provide in the future, but doesn't have priority right now.
With the new user schemas (we are currently working on that), it is possible to define some things on the user, what kind of fields you need and who is allowed to edit.  https://github.com/zitadel/zitadel/issues/6433

We do not have a more specific doc of which permissions are behind a role. the one you found is the default we also use in our cloud.
